lib/action.ml

```diff
@@ -483,7 +491,7 @@ module Action (Github_api : Api.Github) (Slack_api : Api.Slack) (Buildkite_api :
     in
     try%lwt
       let secrets = Context.get_secrets_exn ctx in
-      match%lwt Github.parse_exn headers body ~get_build_branch:(Buildkite_api.get_build_branch ~ctx) with
+      match%lwt Github.parse_exn headers body ~ctx ~api:(module Api_remote.Buildkite) with
```

Comment:
i'm not sure if we want to pass in the whole buildkite module, I've avoided that in my changes as it doesn't feel right to mix the contexts. 

also, check below, we won't get the logs here